components/brave_site_specific/browser/site_specific_script_service_impl.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_site_specific/browser/site_specific_script_service_impl.h"
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_rewards/common/pref_names.h"
+#include "brave/components/brave_site_specific/browser/site_specific_script_config_service.h"
+#include "components/prefs/pref_service.h"
+#include "chrome/browser/profiles/profile.h"
+
+namespace brave_site_specific {
+
+SiteSpecificScriptServiceImpl::SiteSpecificScriptServiceImpl(Profile* profile) :
+    profile_(profile) {
+  DCHECK(!profile_->IsOffTheRecord());
```

Comment:
I'm not sure that this is correct. Won't these scripts also run in incognito profiles?